### PR TITLE
Ensure frontend build process raises errors to sbt

### PR DESCRIPTION
**Pull request by jamespamplin**
Modifed frontend build task in sbt to properly throw errors when build task fails, ensures that builds fail on error.

Other enhancements:
- moved frontend build task to own sbt `AutoPlugin` for easier maintenance
- files can be ignored from build process (such as `hbs` as they're not used client-side yet), improves dev performance when modifying files
- extra build commands can be easily defined
- error output properly captured and will show in browser via Play

### CODE REVIEW

**Comment by susiecoleman** [comment]
I don't know enough about sbt to check this fully but I'm happy to have a look through with someone else

File: project/FrontendBuildPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+    val _sourceDirectory = (sourceDirectory in build in Assets).value
+    val targetDir = (buildOutputDirectory in build in Assets).value
+
+    val cacheDirectory = (streams in Assets).value.cacheDirectory / "npm-build"
+
+    val commands = (buildCommands in build in Assets).value
+
+    def runProcess(cmd: BuildCommand) = {
+      log.info(s"[${cmd.id}] Running `${cmd.command}`")
```
Comment (markjamesbutler) [inline project/FrontendBuildPlugin.scala]: Logback parameters offer superior performance when executing on disabled log.  See http://logback.qos.ch/manual/architecture.html

File: project/FrontendBuildPlugin.scala [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      else
+        Left(FrontendBuildError(cmd.command, exitCode, out.map(_._2).mkString("\n")))
+    }
+
+
+    // use sbt-web compile incremental API to only build when needed
+    val (products, errors) = syncIncremental(cacheDirectory, commands) { ops =>
+      val results = ops.map(op => op -> runProcess(op))
+
+      val opResults = results.map {
```
Comment (markjamesbutler) [inline project/FrontendBuildPlugin.scala]: Add Type as discussed.

**Comment by markjamesbutler** [comment]
:+1 with comment.

**Comment by jamespamplin** [comment]
Removed anonymous types and functions for explicit classes and named functions, and added more comments on whats going on so that things are easier to understand.

**Pull request merged**